server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/TransportForceMergeAction.java

```diff
@@ -102,6 +106,15 @@ protected void shardOperation(
                 .getShard(shardRouting.shardId().id());
             indexShard.ensureMutable(l.map(unused -> indexShard), false);
         }).<EmptyResult>andThen((l, indexShard) -> {
+            boolean forceMergeIsNoOp = indexShard.withEngineException(engine -> {
```

Comment:
drive by comment, but you should make sure you handle when `only_expunge_deletes` is mentioned. That logic check would be different. 

For `only_expunge_deletes`, you should look at the segment infos and verify there aren't any deletes.

For negative number of segments, I think the way to go is to verify if there aren't any "natural merges" that should occur.

Reply:
Thanks very much for your input, @benwtrent! I pushed [d14d3e2](https://github.com/elastic/elasticsearch/pull/135232/commits/d14d3e27c9c04cad40d8dab6772ecf9dce3e317c) to take both `only_expunge_deletes` and a negative number of max segments into account. Could you have a quick look to see if that code matches your expectations?